### DISCUSSION TITLE
Can OpenTelemetry automatically instrument all methods in business code?

### DISCUSSION THREAD

**Discussion by mitht** [open-telemetry/opentelemetry-java-instrumentation#12192] [Q&A] [answered]
Is it possible to automatically instrument and trace all methods in the source code without modifying the business source code?

**Comment by laurit** [comment, marked as answer]
Opentelemetry java agent can be extended with [custom extensions](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension). If you so with you could build an extension that traces all methods based on the package of the class. Usually this is a bad idea, perhaps you should look into profiling. For example java flight recorder provides a low over head profiler that could be used in production.
